lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -78,15 +78,18 @@ def write
               UI.message('- Generating deterministic UUIDs') { project.predictabilize_uuids }
             end
             library_product_types = [:framework, :dynamic_library, :static_library]
-            pod_target_installation_results = @target_installation_results.pod_target_installation_results.values
+
+            pod_target_installation_results = @target_installation_results.pod_target_installation_results
+            results_by_native_target = Hash[pod_target_installation_results.map do |_, result|
+              [result.native_target, result]
+            end]
             project.recreate_user_schemes(false) do |scheme, target|
               next unless target.respond_to?(:symbol_type)
               next unless library_product_types.include? target.symbol_type
-              pod_target_installation_results.each do |pod_target_installation_result|
-                next unless pod_target_installation_result.native_target == target
-                pod_target_installation_result.test_native_targets.each do |test_native_target|
-                  scheme.add_test_target(test_native_target)
-                end
+              installation_result = results_by_native_target[target]
+              next unless installation_result
```

Comment:
is it possible we will never get an installation result? I think its 1-to-1 mapping now (in 1.6.0 master) for every target an installation result is produced.

Reply:
here we can be missing one because we only add the test targets for non-aggregate target installation results